vscode-dotnet-runtime-library/src/IDotnetListVersionsContext.ts

```diff
@@ -0,0 +1,50 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+export interface IDotnetListVersionsContext {
+    /**
+     * @remarks
+     * Holds the parameters necessary to acquire the available SDK or Runtimes.
+     * By available this means only those at the newest minor version for each major version.
+     *
+     * @property listRuntimes - If this is not true (false or undefined), list SDK versions available. Elsewise, list runtimes available.
+     */
+    listRuntimes: boolean | null
+}
+
+/**
+ * @remarks
+ * The result/response from the API to be implemented that returns available SDKs/Runtimes.
+ */
+export type IDotnetListVersionsResult = IDotnetVersion[]
+
+export interface IDotnetVersion {
+    /**
+     * @remarks
+     * Information regarding the version of the .NET SDK / Runtime.
+     *
+     * @property version - The full version of the SDK or Runtime. May include text such as -Preview.
+     * @property channelVersion - The major.minor version.
+     * @property supportPhase - Whether the version is actively in support or in some other stage of support.
+     * @property supportStatus - Is the version in long-term support or 'standard-term' support
+     */
+    version: string,
+    supportStatus: DotnetVersionSupportStatus,
+    supportPhase: DotnetVersionSupportPhase,
+    channelVersion: string
+}
+
+/**
+ * @remarks
+ * lts: Long-term support
+ * sts: Standard-term support
+ */
+export type DotnetVersionSupportStatus = 'lts' | 'sts';
+
+/**
+ * @remarks
+ * preview support does not mean the development is stable and actively supported.
+ */
+export type DotnetVersionSupportPhase = 'active' | 'preview' | 'eol' | 'go-live' | 'maintenance';
```

Comment:
That's a good question. Ideally, we want to discourage people from using EOL dotnets. But: I think we want to expose them because if we don't, it will make it so there's a breaking change in the API every time there's a new EOL SDK. Existing users of the extension seem to hardcode the .NET version they want, so if they rely on this API and then the API suddenly stops providing them a version number, I think that'd be bad. 

I pinged @baronfel about this and he will get back to us on this and the other PM related concerns you mentioned :)

Reply:
We agreed to expose these for now due to the risk of breaking people